### PR TITLE
pscanrules: update reference from http to https

### PR DESCRIPTION
**Pull request by PariharAditya**
Signed-off-by: Aditya Parihar <97739299+PariharAditya@users.noreply.github.com>
## Overview
**fixed from HTTP to HTTPS**
https://www.zaproxy.org/docs/alerts/10050/

## Related Issues
https://github.com/zaproxy/zaproxy/issues/8262

## Checklist
- [ ] Update help
- [x] Update changelog
- [ ] Run `./gradlew spotlessApply` for code formatting
- [ ] Write tests
- [ ] Check code coverage
- [x] Sign-off commits
- [x] Squash commits
- [x] Use a descriptive title

For more details, please refer to the [developer rules and guidelines](https://www.zaproxy.org/docs/developer/dev-rules-and-guidelines/).

### CODE REVIEW

**Comment by kingthorin** [comment]
The CHANGELOG.md should also be updated.

File: addOns/pscanrules/CHANGELOG.md [hunk 1]
```diff
@@ -38,6 +38,10 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/).
 - The CWE for the Cookie Poisoning scan rule was updated to a more specific one.
 - The Strict-Transport-Security Header and Big Redirect scan rules now use alert references for their different types of alerts (Issue 7100).
 
+## [54] - 2024-01-15
+### Changed
+- Update Vulnerabilities references to use https links and retire some which were out-dated (Issue 8262).
+
```
Comment (thc202) [inline addOns/pscanrules/CHANGELOG.md:44]: It should be an entry with `Retrieved from Cache` under `The references for the following scan rules are now all HTTPS (Issue 8262) and in some cases updated:`.

Comment (PariharAditya) [reply addOns/pscanrules/CHANGELOG.md]: Ok ,
Thankyou for the help much needed

Comment (thc202) [reply addOns/pscanrules/CHANGELOG.md]: It should be added under the existing entry though. (The version/date is automatically updated when the add-ons are released.)

Comment (thc202) [reply addOns/pscanrules/CHANGELOG.md]: With existing entry I meant line 9, it just needs to be listed like all the other scan rules.

Comment (PariharAditya) [reply addOns/pscanrules/CHANGELOG.md]: Done.
Thank you for your penance and helping me 🙏

**Review by kingthorin — APPROVED** [review]

**Review by thc202 — APPROVED** [review]

**Comment by thc202** [comment]
Thank you!

**Comment by PariharAditya** [comment]
> Thank you!

🙏🙇‍♂️

**Pull request merged**